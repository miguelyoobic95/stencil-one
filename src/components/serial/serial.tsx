import { Component, Prop, h, JSX} from '@stencil/core';

@Component({
  tag: 'my-serial',
  styleUrl: 'serial.css',
  shadow: true
})
export class MySerialComponent {

  @Prop() value: string = "I am string";

  render() : JSX.Element {
    return (
      <div class="cool">
        <h1>I am a header </h1>
        <p>I am paragraph</p>
        <p>{this.value}</p>
      </div>
    );
  }
}
