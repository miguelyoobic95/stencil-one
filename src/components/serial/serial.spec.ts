import { newSpecPage } from '@stencil/core/testing';
import {serializer} from '../../../test/stencil-serializer';

import { MySerialComponent } from './serial';

describe('serial', () => {

    it('should render', async () => {
        expect.addSnapshotSerializer(serializer)
        const page = await newSpecPage({
            components: [MySerialComponent],
            html: `
                <my-serial value="I am updated value">
                </my-serial>
            `
        });
        debugger;
        console.log('TESTER', serializer.test(page.root))
        // let res = serializer.print(page.root);
        // console.log(res);
        expect(page.root).toMatchSnapshot();
    });
    
});