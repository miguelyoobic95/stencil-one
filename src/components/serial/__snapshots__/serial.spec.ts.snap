// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serial should render 1`] = `
<my-serial value="I am updated value">
  <mock:shadow-root>
    <div class="cool">
      <h1>
        I am a header
      </h1>
      <p>
        I am paragraph
      </p>
      <p>
        I am updated value
      </p>
    </div>
  </mock:shadow-root>
</my-serial>
`;
